### PR TITLE
Add a few tests for linodes index page

### PR DESCRIPTION
**Pull request by SirCmpwn**

### CODE REVIEW

File: src/linodes/components/Linode.js [hunk 1]
```diff
@@ -29,8 +29,8 @@ function renderCard(props) {
   const selectedClass = isSelected ? 'selected' : '';
 
   const checkbox = isSelected ?
-    <input type="checkbox" checked="checked" onClick={select} /> :
-    <input type="checkbox" onClick={select} />;
+    <input type="checkbox" checked="checked" onChange={select} /> :
+    <input type="checkbox" onChange={select} />;
```
Comment (eatonphil) [inline src/linodes/components/Linode.js]: +1

**Comment by SirCmpwn** [comment]
Travis is insane. Tests pass and linter is good locally. Merge advised.

File: test/linodes/layouts/IndexPage.spec.js [hunk 2]
```diff
@@ -21,11 +22,11 @@ describe('linodes/layouts/IndexPage', () => {
   const linodes = {
     pagesFetched: [0],
     totalPages: 1,
-    linodes: [
-      testLinode,
-      { ...testLinode, id: 'linode_1235' },
-      { ...testLinode, id: 'linode_1236', state: 'offline' },
-    ],
+    linodes: {
```
Comment (eatonphil) [inline test/linodes/layouts/IndexPage.spec.js]: What happened here?

Comment (SirCmpwn) [inline test/linodes/layouts/IndexPage.spec.js]: I screwed up the layout of this object when I wrote it in the first place. This is the correct state.

File: test/linodes/layouts/IndexPage.spec.js [hunk 3]
```diff
@@ -59,7 +60,55 @@ describe('linodes/layouts/IndexPage', () => {
...
+      expect(fetchStub.firstCall.args[1]).to.equal(`/linodes/linode_1234${endpoint}`);
+    };
+  }
+
+  [
+    ['reboots selected linodes when reboot is pressed', 'Reboot', '/reboot'],
+    ['shuts down selected linodes when power off is pressed', 'Power Off', '/shutdown'],
+    ['boots selected linodes when boot is pressed', 'Power On', '/boot'],
+    ['deletes selected linodes when deletes is pressed', 'Delete', ''],
+  ].map(n => it(n[0], makePowerTest(n[1], n[2])));
```
Comment (eatonphil) [inline test/linodes/layouts/IndexPage.spec.js]: In general, I'd prefer: `map([description, dropdownLabel, endpoint] => ...)`

Comment (SirCmpwn) [inline test/linodes/layouts/IndexPage.spec.js]: map isn't a global in JS

Comment (eatonphil) [inline test/linodes/layouts/IndexPage.spec.js]: I meant destructuring the arguments.

Comment (SirCmpwn) [inline test/linodes/layouts/IndexPage.spec.js]: Ah, I see. Sure, I'll make that change.

**Comment by SirCmpwn** [comment]
Let's see if Travis gets its shit right this time.

**Comment by eatonphil** [comment]
Thanks. I guess we'll see if tests pass this time. Weird that this is triggering both pr and push builds. Perhaps it's because you are developing directly on this repo instead of on a fork?

**Comment by SirCmpwn** [comment]
Yeah, it's one build for the PR and one build for the branch. I could move to a fork, I suppose.

**Comment by eatonphil** [comment]
Might cut wait time in half.

**Comment by SirCmpwn** [comment]
Same result. Does not make any sense. Suggesting we move forward with merge.

**Comment by SirCmpwn** [comment]
I filed a bug with travis ftr

**Comment by eatonphil** [comment]
Saw that. Alright, merging.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**